homeassistant/components/knx/climate.py

```diff
@@ -322,6 +354,43 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
             )
             self.async_write_ha_state()
 
+    @property
+    def fan_mode(self) -> str:
+        """Return the fan setting."""
+
+        fan_speed = self._device.current_fan_speed
+
+        if fan_speed is None:
+            return FAN_OFF
+
+        if fan_speed == 0:
+            return FAN_OFF
+
+        if self._attr_fan_modes is None:
+            return FAN_OFF
+
+        if self._device.fan_mode == FanSpeedMode.STEP:
+            return self._attr_fan_modes[fan_speed]
+
+        return f"{fan_speed}%"
```

Comment:
I think you want the last value here, no?

Reply:
anyway, I fixed the code to find the closest and added a test for that.

Reply:
Nice idea 👍
One suggestion though: I'd remove `0` from `self._fan_modes_percentages` in the `fan_modes` property. That way if modes are `[0, 33, 66, 100]` and the controller returns `15%` (or lower >0), it would show `FAN_LOW` instead of `FAN_OFF` and the user would have the ability to turn the fan off completely as the OFF item wouln't already be preselected.